apps/framr-web/lib/services/core/modules/tools/ToolsService.ts

```diff
@@ -73,10 +69,10 @@ export class ToolsService implements ToolInterface {
   findAll(): void {
     let channel = ToolsEventChannel.FIND_ALL_TOOLS_CHANNEL;
     this.database
-      .findAll('tools')
+      .findAll(this.STORE_NAME)
       .then((result) => {
         this.eventBus.emit(channel, {
-          data: { ...result },
+          data: { result },
```

Comment:
data must be an array of type tools not an object, if the result is not of type Tool[], use it build the necessary type we are looking for

Reply:
I don't understand to the extent that findAll finds elements response will already be an array otherwise response will have a null value and it is this response value which will be sent emit in either case